src/components/Documentation/RightPanel/index.tsx

```diff
@@ -133,7 +162,11 @@ const RightPanel: React.FC<IRightPanelProps> = ({
           </div>
           <div className={styles.contentBlock} ref={contentBlockRef}>
             {headings.map(({ slug, text }) => (
-              <div id={`link-${slug}`} key={`link-${slug}`}>
+              <div
+                id={generateHeadingId(slug)}
+                key={generateHeadingId(slug)}
+                ref={tableOfContentsRefs[generateHeadingId(slug)]}
+              >
```

Comment:
It'd be better to only run `generateHeadingId` only once, even if it means we need to use the uglier curly-brace syntax.

Reply:
Nice catch! Updated in https://github.com/kangzeroo/dvc.org/pull/1/commits/faff75314c86bf59687e66883f50be045c97f0de